### PR TITLE
Configure Dependabot to ignore patch updates

### LINKED ISSUE #6849: Configure Dependabot to exclude 'patch' version updates
### Overview
The Dependabot is currently configured to check for updated dependencies, and if a new version is available to create a PR. The current config checks for all version updates regardless of size. We need to modify the configuration file to reduce the frequency of the Dependabot updates and exclude 'patch' version update.

### Action Items
Reference the "Resources/ Instructions" below for information regarding Semver and Dependabot version updates.
- [x] Locate the [dependabot.yml](https://github.com/hackforla/website/blob/gh-pages/.github/dependabot.yml) file
- [x] Modify the configuration file as needed so that the Dependabot ignores Semver 'patch' version updates, but continues to generate PRs for 'major' and 'minor' updates
- [x] Add ignore statement ( IMPORTANT NOTE this syntax needs to be verified)
```
ignore:
   update-types: ["version-update:semver-patch"]
```  
- [ ] Important note: confirm that this **does not apply to security updates and patches**, only to version updates.

### Resources/Instructions
- [Introduction to Semantic Versioning](https://www.geeksforgeeks.org/introduction-semantic-versioning/) from Geeks for Geeks
- [About Dependabot version updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates)
- [Configuring Dependabot version updates](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates#enabling-dependabot-version-updates)  

This issue was initiated by ER #6797

### PR DESCRIPTION
**Pull request by JackCasica**
Fixes #6849

### What changes did you make?
  - Modified the Dependabot configuration to ignore semver patch updates.

### Why did you make the changes (we will use this info to test)?
  - We are ignoring patch updates in dependencies to reduce noise from minor updates and focus on more significant changes to dependencies. Security patches are not ignored.

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
 - Not applicable.

### CODE REVIEW

**Comment by A-Wu5** [comment]
Availability: T-Th 7-10pm
ETA: Th

**Comment by iancooperman** [comment]
ETA: Sat 5/25 EoD
Availability: Wed 5/22 6pm-9pm, Thurs 5/23 N/A, Fri 5/24 5pm-9pm, Sat 5/25 10am-9pm

**Review by t-will-gillis — CHANGES_REQUESTED** [review]
Hi @JackCasica @A-Wu5 and @iancooperman: I started looking at testing for the Dependabot changes and I had it in my head this would be easier to test. Here is what I found out:
- I found references in Stack Overflow to actions other people wrote to test the Dependabot updates but long story short I did not find anything useful.
- So I created a couple of test branches and reverted the `stefanzweifel/git-auto-commit-action` from version 5.0.1 to 5.0.0. Then branch by branch I Enabled and Disabled the Dependabot version updates
  - (If you are curious, go to "Settings", then under "Security" at left select "Code Security and Analysis", then Enable [and later Disable] the "Dependabot version updates")
  - (You can also go to "Insights", then "Dependency Graph" to the left, then select the tab for "Dependabot" and all of the previous runs will be shown, and you can trigger a new run)
- Every time the `dependabot.yml` file is touched, the Dependaot runs: 
  - I ran the Dependabot under the existing code, and the PR was generated to update `stefanz..` to 5.0.1, as expected.
  - Then edited the other branch with the proposed updates to `dependabot.yml`- and received the error shown here:  
  

  <details>
  Screenshot 2024-05-24 191636


  </details>
   

TL;DR:
I believe that the ignore statement needs to know which of the 'dependency-name' s to 'ignore'. If you add in a wildcard the error goes away and this seems to work:  
  
  ```  
    ignore:
      - dependency-name: "*"
        update-types: ["version-update:semver-patch"]
  ```  

When I substituted this into `dependabot.yml`, the Dependabot did not try to update `stefanz...` so I believe the code needs to change as shown. 

Thanks @JackCasica for working on this!

**Comment by A-Wu5** [comment]
Hi @t-will-gillis, what did you mean by reverting `stefanzweifel/git-auto-commit-action`?

**Comment by t-will-gillis** [comment]
Hey @A-Wu5 

>Hi @t-will-gillis, what did you mean by reverting stefanzweifel/git-auto-commit-action?

Apologies, I left out some information! What I meant is: in order to test whether this PR successfully ignores patch updates, we could reset a dependency that we know has a recent patch update. The workflow [schedule-daily-1100](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/schedule-daily-1100.yml) was [recently updated](https://github.com/hackforla/website/pull/6785) to `stefanzweifel/git-auto-commit-action@v5.0.1`, so we can reset this to `stefanzweifel/git-auto-commit-action@v5.0.0` to see whether the Dependabot does or does not create another PR to update this dependency.

**Comment by A-Wu5** [comment]
I see, thank you for the clarification @t-will-gillis. That makes sense.

**Review by A-Wu5 — CHANGES_REQUESTED** [review]
Hi @JackCasica, thanks for working on this issue. I created a new branch with your changes in my local repo and got the same error message as @t-will-gillis stated above:

```
The property '#/updates/0/ignore/0' did not contain a required property of 'dependency-name'
The property '#/updates/1/ignore/0' did not contain a required property of 'dependency-name'
```

Please make the necessary changes as @t-will-gillis mentioned above.

**Comment by iancooperman** [comment]
@JackCasica After getting the same error myself, I concur with @A-Wu5 and @t-will-gillis.
image

**Review by iancooperman — CHANGES_REQUESTED** [review]
Please refer to my previous comment regarding the changes needed.
https://github.com/hackforla/website/pull/6870#issuecomment-2132600655

**Comment by JackCasica** [comment]
Hi @t-will-gillis, @A-Wu5, @iancooperman ~

Thanks for all of the feedback and testing. I have made the requested changes to the `dependabot.yml` file. Specifically, I added the `dependency-name: "*"` to the ignore statements to ensure that the configuration works as expected. 

The updated ignore configurations for both package ecosystems now looks like this:
```yaml
ignore:
  - dependency-name: "*"
    update-types: ["version-update:semver-patch"]
```

**Review by t-will-gillis — APPROVED** [review]
Hey @JackCasica Perfect, thanks for making the changes!

**Review by A-Wu5 — APPROVED** [review]
Looks good to me. Great job @JackCasica!

**Review by iancooperman — APPROVED** [review]
Looks great. Nice job, @JackCasica!
